spark/v3.3/build.gradle

```diff
@@ -62,6 +62,7 @@ project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}") {
     implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}")
 
     compileOnly "com.google.errorprone:error_prone_annotations"
+    compileOnly "com.github.ben-manes.caffeine:caffeine"
```

Comment:
These changes look incorrect. Why is this new compile dependency needed when there is no code change?

Reply:
there is a change in GenericReader that could reuse positionalDeletes into between the tasks from the same split